packages/core/src/utils/translation.ts

```diff
@@ -0,0 +1,39 @@
+import { Translation } from '@logto/schemas';
+
+export const isValidStructure = (
+  fullTranslation: Translation,
+  partialTranslation: Partial<Translation>
```

Comment:
For the usage, you can not ensure that this parameter is `Partial<Translation>`, maybe change to a nested record of string.